src/navbar/menu.tsx

```diff
@@ -1,11 +1,12 @@
 import React from "react";
 import { Link, MenuItem, MobileWrapper, Wrapper } from "./menu.styles";
-import { NavItemCommonProps } from "./types";
+import { NavItemCommonProps, NavItemProps } from "./types";
 
 interface Props<T> {
-    items: NavItemCommonProps<T>[];
+    items: NavItemProps<T>[];
```

Comment:
Non-breaking change to Navbar, allowing both links and JSX.Element to be passed into the subMenu. 
this is to correctly display the profile name. And subsequently changes in future ticket [MOL-19016](https://sgtechstack.atlassian.net/browse/MOL-19016)

<img width="206" height="239" alt="image" src="https://github.com/user-attachments/assets/43f8a7de-4071-4514-8093-f51ab7800c40" />

Reply:
Note that there should not be a gap between the fullsize navbar and submenu:
<img width="369" height="457" alt="image" src="https://github.com/user-attachments/assets/de1c9ab3-87b3-4c98-8160-80246fe1b542" />

Reply:
menu changes to be tackled separately